### ISSUE TITLE
Fast CSV interface for nmatrix

### ISSUE THREAD

**Issue by v0dro** [SciRuby/nmatrix#407]
We could something like [fastest-csv](https://github.com/brightcode/fastest-csv) and load data directly from the C interface, no ruby involved. That should be real fast.

**Comment by v0dro** [comment]
Or maybe even use [libcsv](http://sourceforge.net/projects/libcsv/).

**Comment by mohawkjohn** [comment]
I think that'd be awesome.

**Comment by khalibartan** [comment]
@v0dro What do we want in this issue ? Means should I try to write a wrapper for libcsv to work with nmatrix or, should I implement a CSV reader for nmatrix itself.

**Comment by v0dro** [comment]
I'd prefer libcsv. No point in reinventing the wheel.

**Comment by mohawkjohn** [comment]
I agree. It'd be better not to introduce an additional dependency. Perhaps there could be a pure Ruby csv reader as well for it to fall back on.

Here's where such a thing would go: https://github.com/SciRuby/nmatrix/tree/master/lib/nmatrix/io

**Comment by saru95** [comment]
Is this being worked upon ? I'd like to write a ruby wrapper.

**Comment by mohawkjohn** [comment]
I don't believe so.

**Comment by v0dro** [comment]
Have a look at this: https://github.com/wiseio/paratext

I think its the fastest CSV reader around and having a wrapper over that would be awesome :dancing_women:
